main.tf

```diff
@@ -1 +1,89 @@
+resource "azurerm_public_ip" "this" {
+  count               = var.public_ip_config.enabled ? 1 : 0
+  name                = local.public_ip.name
+  resource_group_name = var.resource_group_name
+  location            = var.virtual_machine_config.location
+  allocation_method   = var.public_ip_config.allocation_method
 
+  lifecycle {
+    ignore_changes = [
+      tags
+    ]
+  }
+}
+
+resource "azurerm_network_interface" "this" {
+  name                          = local.nic.name
+  location                      = var.virtual_machine_config.location
+  resource_group_name           = var.resource_group_name
+  dns_servers                   = var.nic_config.dns_servers
+  enable_accelerated_networking = var.enable_accelerated_networking
+
+  ip_configuration {
+    name                          = local.nic.ip_config_name
+    subnet_id                     = var.subnet.id
+    private_ip_address_allocation = var.nic_config.private_ip == null ? "Dynamic" : "Static"
+    private_ip_address            = var.nic_config.private_ip
+    public_ip_address_id          = var.public_ip_config.enabled ? azurerm_public_ip.this[0].id : null
+  }
+
+  lifecycle {
+    ignore_changes = [
+      tags
+    ]
+  }
+}
+
+resource "azurerm_network_interface_security_group_association" "this" {
+  count = var.nic_config.nsg != null ? 1 : 0
+  network_interface_id = azurerm_network_interface.this.id
+  network_security_group_id = var.nic_config.nsg.id
+}
+
+resource "azurerm_linux_virtual_machine" "this" {
+  name                            = local.virtual_machine.name
+  computer_name                   = var.virtual_machine_config.hostname
+  location                        = var.virtual_machine_config.location
+  resource_group_name             = var.resource_group_name
+  size                            = var.virtual_machine_config.size
+  provision_vm_agent              = true
```

Comment:
this already defaults to "true". This line is unnecessary. 
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine#provision_vm_agent

Reply:
Resolved as recommend.